api/src/test/java/org/openmrs/module/attachments/obs/ImageAttachmentHandlerIT.java

```diff
@@ -39,6 +41,11 @@ public void saveComplexData_shouldSaveThumbnailToDisk() throws IOException {
 		Assert.assertThat(thumbnail.length(), lessThan(file.length()));
 		BufferedImage img = ImageIO.read(thumbnail);
 		Assert.assertEquals(ImageAttachmentHandler.THUMBNAIL_MAX_HEIGHT, Math.max(img.getHeight(), img.getWidth()));
+		
+		// Now check rename the file by appending NO_THUMBNAIL_SUFFIX to the file
+		String fileName = ImageAttachmentHandler.saveThumbnailOrRename(file, img.getHeight(), img.getWidth());
+		Assert.assertTrue(
+		    StringUtils.endsWith(FilenameUtils.removeExtension(fileName), ImageAttachmentHandler.NO_THUMBNAIL_SUFFIX));
```

Comment:
This is a misleading test IMO, but what you could do is test that the file behind the file name marked as 'no thumbnail' doesn't exist:
```java
File noThumbnailFile = new File(testHelper.getComplexObsDir() + "/" +
    ImageAttachmentHandler.buildNoThumbnailFileName(mpFile.getOriginalFilename()));
Assert.assertFalse(noThumbnailFile.exists());
```